### ISSUE TITLE
heap corruption causing a crash

### ISSUE THREAD

**Issue by davidnich** [qorelanguage/qore#1184]
cause unknown - stack trace:

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000032f340eb6d in raise () from ../rhel5-libs-6/lib64/libpthread.so.0
[Current thread is 1 (Thread 0x2b33c5a09940 (LWP 16066))]
(gdb) bt
#0  0x00000032f340eb6d in raise () from ../rhel5-libs-6/lib64/libpthread.so.0
#1  0x00002b3389f285b6 in skgesigOSCrash () from ../rhel5-libs-6/appl/oracle/product/client_11202/lib/libclntsh.so.11.1
#2  0x00002b338a1c6609 in kpeDbgSignalHandler () from ../rhel5-libs-6/appl/oracle/product/client_11202/lib/libclntsh.so.11.1
#3  0x00002b3389f287c6 in skgesig_sigactionHandler () from ../rhel5-libs-6/appl/oracle/product/client_11202/lib/libclntsh.so.11.1
#4  <signal handler called>
#5  0x00000032f2c71b91 in _int_malloc () from ../rhel5-libs-6/lib64/libc.so.6
#6  0x00000032f2c73dfe in malloc () from ../rhel5-libs-6/lib64/libc.so.6
#7  0x00002b33860eede8 in operator new (sz=sz@entry=24) at ../../../../libstdc++-v3/libsupc++/new_op.cc:50
#8  0x00002b3385b0d78a in __gnu_cxx::new_allocator<std::_List_node<HashMember*> >::allocate (this=0x2b33b9ee1310, __n=1) at /opt/gcc-5.4.0/include/c++/5.4.0/ext/new_allocator.h:104
#9  std::__cxx11::_List_base<HashMember*, std::allocator<HashMember*> >::_M_get_node (this=0x2b33b9ee1310) at /opt/gcc-5.4.0/include/c++/5.4.0/bits/stl_list.h:392
#10 std::__cxx11::list<HashMember*, std::allocator<HashMember*> >::_M_create_node (this=0x2b33b9ee1310, __x=<synthetic pointer>) at /opt/gcc-5.4.0/include/c++/5.4.0/bits/stl_list.h:553
#11 std::__cxx11::list<HashMember*, std::allocator<HashMember*> >::_M_insert (__x=<synthetic pointer>, __position=..., this=0x2b33b9ee1310) at /opt/gcc-5.4.0/include/c++/5.4.0/bits/stl_list.h:1754
#12 std::__cxx11::list<HashMember*, std::allocator<HashMember*> >::push_back (__x=<synthetic pointer>, this=0x2b33b9ee1310) at /opt/gcc-5.4.0/include/c++/5.4.0/bits/stl_list.h:1089
#13 qlist<HashMember*>::push_back (t=<synthetic pointer>, this=0x2b33b9ee1310) at ../include/qore/qlist:102
#14 qore_hash_private::findCreateMember (this=0x2b33b9ee1310, key=0x2b3385b40c76 "file") at ../include/qore/intern/QoreHashNodeIntern.h:137
#15 0x00002b33859fef79 in hash_assignment_priv::hash_assignment_priv (must_already_exist=false, key=0x2b3385b40c76 "file", n_h=..., this=0x2b33c5a059d0) at QoreHashNode.cpp:843
#16 QoreHashNode::setKeyValue (this=this@entry=0x2b33b9a3ffd0, key=key@entry=0x2b3385b40c76 "file", val=val@entry=0x2b33b9b98bf0, xsink=xsink@entry=0x0) at QoreHashNode.cpp:175
#17 0x00002b3385a0026d in QoreException::makeExceptionObject (this=this@entry=0x2b33b98a4330) at QoreException.cpp:77
#18 0x00002b3385a85c3e in TryStatement::execImpl (this=0x1b8fdf70, return_value=..., xsink=0x2b33c5a08fb0) at TryStatement.cpp:83
#19 0x00002b338598a343 in AbstractStatement::exec (this=0x1b8fdf70, return_value=..., xsink=xsink@entry=0x2b33c5a08fb0) at AbstractStatement.cpp:54
#20 0x00002b33859d07f1 in StatementBlock::execIntern (this=this@entry=0x1b8fe3e0, return_value=..., xsink=xsink@entry=0x2b33c5a08fb0) at StatementBlock.cpp:221
#21 0x00002b3385a87630 in StatementBlock::execImpl (this=0x1b8fe3e0, return_value=..., xsink=0x2b33c5a08fb0) at StatementBlock.cpp:203
#22 0x00002b3385a8624b in StatementBlock::exec (xsink=0x2b33c5a08fb0, this=0x1b8fe3e0) at StatementBlock.cpp:168
#23 UserVariantBase::evalIntern (this=this@entry=0x1b8fe4e0, argv=..., self=self@entry=0x2b33bd88f770, xsink=xsink@entry=0x2b33c5a08fb0) at Function.cpp:1313
#24 0x00002b3385a8c6be in UserVariantBase::eval (this=this@entry=0x1b8fe4e0, name=0x1b8fe620 "pollData", ceh=ceh@entry=0x2b33c5a05dc0, self=self@entry=0x2b33bd88f770, xsink=xsink@entry=0x2b33c5a08fb0, qc=0x1b8fc900)
    at Function.cpp:1365
...
```

The crash happened while trying to create a standard exception in `WebSocketHandler.qm:105` due to an operational timeout (`QUEUE-TIMEOUT`).

The source of the heap corruption bug is currently unknown.

**Comment by davidnich** [comment]
possibly fixed by the reference-counting fixes in #1268, #1270 and #1271 - TBE

**Comment by davidnich** [comment]
root cause #1084
